povm_toolbox/quantum_info/base_povm.py

```diff
@@ -13,63 +13,33 @@
 from __future__ import annotations
 
 from abc import ABC, abstractmethod
-from typing import Any
+from typing import TypeVar
 
 import numpy as np
-from qiskit.quantum_info import DensityMatrix, Operator
+from qiskit.quantum_info import DensityMatrix, SparsePauliOp, Statevector
 
+from .base_dual import BaseDUAL
+from .base_frame import BaseFrame
 
-class BasePOVM(ABC):
-    """Abstract base class that contains all methods that any specific POVM subclass should implement."""
+# type of the labels used to specify povm effects
+LabelT = TypeVar("LabelT")
 
-    @property
-    @abstractmethod
-    def dimension(self) -> int:
-        """Give the dimension of the Hilbert space on which the effects act."""
 
-    @property
-    def n_subsystems(self) -> int:
-        """Return the number of subsystems which the effects act on.
+class BasePOVM(BaseFrame[LabelT], ABC):
```

Comment:
As a concrete example, I think this should work:
```suggestion
class BasePOVM(BaseFrame, ABC):
```
Iirc this works and implies that `BasePOVM` is also `Generic`

Reply:
It didn't work this way, I either had to add a generic at the end 
```python
class BasePOVM(BaseFrame, ABC, Generic[LabelT]):
```
or keep it like 
```python
class BasePOVM(BaseFrame[LabelT], ABC):
```
, which is what I did.

However it worked fine to import `LabelT` instead of re-defining it 👍